### PR TITLE
fix(material-experimental): unit tests getting wrong root element and add API for getting property value

### PR DESCRIPTION
**Pull request by crisbeto**
Based on the discussion in https://github.com/angular/components/pull/16697#discussion_r311208562, these changes fix a couple of things that I ran into while doing the `mat-autocomplete` test harness in https://github.com/angular/components/pull/16620.

* Fixes querying for elements outside the harness not working, because the wrong root node was set.
* Adds an API to retrieve the value of a property on a DOM node.

**Note:** marking as P2, because the issue with the root node is something that'll come up every time we write a harness for an overlay-based component.

### CODE REVIEW

File: src/cdk-experimental/testing/protractor/protractor-element.ts [hunk 1]
```diff
@@ -55,4 +55,8 @@ export class ProtractorElement implements TestElement {
     const classes = (await this.getAttribute('class')) || '';
     return new Set(classes.split(/\s+/).filter(c => c)).has(name);
   }
+
+  async getPropertyValue(name: string): Promise<any> {
```
Comment (mmalerba) [inline src/cdk-experimental/testing/protractor/protractor-element.ts]: Why is this necessary? `getAttributeValue` should fall back to getting the property

Comment (devversion) [reply src/cdk-experimental/testing/protractor/protractor-element.ts]: This is what I had as commit description when I did the same changes on another PR:

```
Most of the time harnesses are able to use `getAttribute` to retrieve data
from DOM elements, but there are edge-cases where it's not correct to
retrieve data through the attribute.

For example: Consider an `input` element which has a `value` attribute set to
the initial input value. Now when the value of the input changes through
interaction, the `value` attribute is not refreshed as it's a static attribute.

In that case, the harness should retrieve the value by reading the DOM element
property. This is needed in order to implement the `MatInputHarness`.
```

Comment (mmalerba) [reply src/cdk-experimental/testing/protractor/protractor-element.ts]: Ah I see, does it still make sense to have `getAttribute` fallback to getting the property then? Considering that we have a separate method for getting the property now. 

The decision to have the fallback behavior was part of the original internal harness framework, I'm assuming because that's how protractor's getAttribute method works.

@jelbourn do you have an opinion?

Comment (devversion) [reply src/cdk-experimental/testing/protractor/protractor-element.ts]: @mmalerba Yeah IIRC protractor by default also falls back to the property value. I could see the point of `getAttribute` only checking the attribute though. 

FWIW through, it was in some cases convenient that it falls back to the property, but we could implement that on our own if needed.

Comment (mmalerba) [reply src/cdk-experimental/testing/protractor/protractor-element.ts]: Yeah I can see why we might want the fallback behavior. In most cases users probably don't care which it is, `value` is kind of a special case

Comment (jelbourn) [reply src/cdk-experimental/testing/protractor/protractor-element.ts]: I would lean towards special-casing attributes where we know this doesn't work, but we can discuss at the next team meeting

Comment (jelbourn) [inline src/cdk-experimental/testing/protractor/protractor-element.ts]: Could these APIs use `unknown` instead of `any`?

Comment (devversion) [reply src/cdk-experimental/testing/protractor/protractor-element.ts]: Good idea. I'd also vote for that.

Comment (mmalerba) [reply src/cdk-experimental/testing/protractor/protractor-element.ts]: By "special-casing" do you mean preferring the property over the attribute in the case of properties like `value`? If so, is there a chance that could prevent a user from accessing the `value` attribute when that really is what they want to do?

**Comment by mmalerba** [comment]
re-apply merge ready after comments are resolved

**Comment by jelbourn** [comment]
Commenting here for posterity: decided in the team meeting to do separate methods for attributes and properties with no fallbacks

File: src/cdk-experimental/testing/protractor/protractor-element.ts [hunk 2]
```diff
@@ -123,7 +123,7 @@ export class ProtractorElement implements TestElement {
   }
 
   async getAttribute(name: string): Promise<string|null> {
-    return this.element.getAttribute(name);
+    return browser.executeScript(`return arguments[0].getAttribute('${name}')`, this.element);
```
Comment (mmalerba) [inline src/cdk-experimental/testing/protractor/protractor-element.ts]: I think its better to pass `name` as a second argument in case it contains `'`

File: src/cdk-experimental/testing/protractor/protractor-element.ts [hunk 3]
```diff
@@ -136,4 +136,8 @@ export class ProtractorElement implements TestElement {
     const {x: left, y: top} = await this.element.getLocation();
     return {width, height, left, top};
   }
+
+  async getPropertyValue(name: string): Promise<any> {
+    return browser.executeScript(`return arguments[0]['${name}']`, this.element);
```
Comment (mmalerba) [inline src/cdk-experimental/testing/protractor/protractor-element.ts]: pass name as a `second` argument

File: src/cdk-experimental/testing/protractor/protractor-element.ts [hunk 4]
```diff
@@ -136,4 +136,8 @@ export class ProtractorElement implements TestElement {
     const {x: left, y: top} = await this.element.getLocation();
     return {width, height, left, top};
   }
+
+  async getPropertyValue(name: string): Promise<any> {
```
Comment (mmalerba) [inline src/cdk-experimental/testing/protractor/protractor-element.ts]: Let's call it `getProperty` so the name matches `getAttribute`

**Comment by crisbeto** [comment]
@mmalerba all the feedback has been addressed.
